engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2174,6 +2178,33 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
 
   }
 
+  private Collection<PTOperator> getUnifiersInCheckpointGroup(Set<OperatorMeta> checkpointGroup, Collection<PTOperator> operators)
+  {
+    boolean isDelayPartOfGroup = false;
+    for (OperatorMeta om : checkpointGroup) {
+      if (om.getOperator() instanceof Operator.DelayOperator) {
+        isDelayPartOfGroup = true;
+      }
+    }
+    // Include unifiers in group if delay operator is present
+    // Do only if group contains a delay operator
+    if (isDelayPartOfGroup) {
```

Comment:
Sorry, I misunderstood your comment earlier. In absence of delay, what is the problem if unifier is part of a separate checkpoint group? This is what happens currently and seems to be working okay.